pandas/core/internals.py

```diff
@@ -1795,6 +1797,99 @@ def _unstack(self, unstacker_func, new_columns):
         return blocks, mask
 
 
+class ExtensionBlock(NonConsolidatableMixIn, Block):
+    """Block for holding extension types.
+
+    Notes
+    -----
+    This is the holds all 3rd-party extension types. It's also the immediate
+    parent class for our internal extension types' blocks, CategoricalBlock.
+
+    ExtensionArrays are limited to 1-D.
+    """
+    is_extension = True
+    is_bool = False
+
+    def __init__(self, values, placement, ndim=None):
+        self._holder = type(values)
+        super(ExtensionBlock, self).__init__(values, placement, ndim=ndim)
+
+    def get_values(self, dtype=None):
+        # ExtensionArrays must be iterable, so this works.
+        values = np.asarray(self.values)
+        if values.ndim == self.ndim - 1:
+            values = values.reshape((1,) + values.shape)
+        return values
+
+    def to_dense(self):
+        return self.values.get_values()
+
+    def take_nd(self, indexer, axis=0, new_mgr_locs=None, fill_tuple=None):
+        """
+        Take values according to indexer and return them as a block.
+        """
+        if fill_tuple is None:
+            fill_value = None
+        else:
+            fill_value = fill_tuple[0]
+
+        # axis doesn't matter; we are really a single-dim object
+        # but are passed the axis depending on the calling routing
+        # if its REALLY axis 0, then this will be a reindex and not a take
+        new_values = self.values.take(indexer, fill_value=fill_value)
+
+        # if we are a 1-dim object, then always place at 0
+        if self.ndim == 1:
+            new_mgr_locs = [0]
+        else:
+            if new_mgr_locs is None:
+                new_mgr_locs = self.mgr_locs
+
+        return self.make_block_same_class(new_values, new_mgr_locs)
+
+    def _can_hold_element(self, element):
+        # XXX:
+        # Not defined on NCM.
+        # Categorical got True from ObjectBlock
+        # DatetimeTZ gets DatetimeBlock
+        # Sparse gets Block
+        # Let's just assume yes for now, but we can maybe push
+        # this onto the array.
+        return True
+
+    def convert(self, copy=True, **kwargs):
```

Comment:
let' start much more minimally and add methods individually, rather than the reverse (of adding everything).

Reply:
This is copy pasted from CategoricalBlock, which now inherits from this one. So we need to keep it in either places to not change anything. You mean keep it in CategoricalBlock?

But, I think this one is actually the same as base Block, and if we leave it out here is just inherited from there. So think we can remove it.

Reply:
Removed.